src/scripts/tagImporter.ts

```diff
@@ -11,28 +11,52 @@ const DATADIR = `${
   process.env.HOME
 }/gnl/data.detroitledger.org/profiles/gnl_profile/exporters`;
 
-const createBasicParser = (model: Sequelize.Model<any, any>) => (err, data) =>
-  data.map(datum => model.create(datum));
-
-[
-  {
-    file: 'grantTag.csv',
-    parser: createBasicParser(db.GrantTag),
-  },
-  {
-    file: 'organizationTag.csv',
-    parser: createBasicParser(db.OrganizationTag),
-  },
-  {
-    file: 'nteeGrantType.csv',
-    parser: createBasicParser(db.NteeGrantType),
-  },
-  {
-    file: 'nteeOrganizationType.csv',
-    parser: createBasicParser(db.NteeOrganizationType),
-  },
-].map(cfg =>
-  fs
-    .createReadStream(`${DATADIR}/${cfg.file}`)
-    .pipe(csvParse({ delimiter: ',', columns: true }, cfg.parser))
-);
+function doImport() {
+  const promises = [
+    {
+      file: 'grantTag.csv',
+      model: db.GrantTag,
+    },
+    {
+      file: 'organizationTag.csv',
+      model: db.OrganizationTag,
+    },
+    {
+      file: 'nteeGrantType.csv',
+      model: db.NteeGrantType,
+    },
+    {
+      file: 'nteeOrganizationType.csv',
+      model: db.NteeOrganizationType,
+    },
+  ].map(
+    cfg =>
+      new Promise((resolve, reject) => {
+        const stream = fs
+          .createReadStream(`${DATADIR}/${cfg.file}`)
+          .pipe(csvParse({ delimiter: ',', columns: true }));
+
+        stream.on('readable', async () => {
+          let chunk;
+
+          while (null !== (chunk = stream.read())) {
```

Comment:
This works great, and all good. As a more general style note, I personally prefer having super basic conditionals because I find them easier to read, even if it adds another line:

```
let chunk = stream.read();

while (null !== chunk) {
  await cfg.model.create(chunk);
  chunk = stream.read();
}
```

Reply:
good point. i was using the idiomatic example from the node docs, but this syntax is dumb!